packages/slate/src/models/node.js

```diff
@@ -958,18 +925,29 @@ class Node {
   }
 
   /**
-   * Return a set of all keys in the node.
+   * Return a set of all keys in the node as an array.
    *
-   * @return {Set<String>}
+   * @return {Array<String>}
    */
 
-  getKeys() {
+  getKeysAsArray() {
```

Comment:
I think this method can also be improved in performance if it was recursive, such that each nested child's keys would be cached, instead of having to iterate the entire tree each time the document changes.